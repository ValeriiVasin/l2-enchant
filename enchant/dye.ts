import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { mergeEnchantMaps } from '../helpers/merge-enchants';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const dyeEnchantMap = mergeEnchantMaps([
  createEnchantMap({
    strategy: EnchantStrategy.Safe,
    enchantItem: Item.DyePowder,
    chances: [50, 25, 20, 15, 10, 8, 6, 5, 5, 5, 5],
    items: itemsRange(Item.Dye, Item.Dye_11),
  }),
  createEnchantMap({
    strategy: EnchantStrategy.Safe,
    enchantItem: Item.EnhancedDyePowder,
    chances: [10, 8, 6, 5, 4, 3, 2, 1.5, 0.5],
    items: itemsRange(Item.Dye_11, Item.Dye_20),
  }),
]);
