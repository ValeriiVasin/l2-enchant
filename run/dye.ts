import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.Dye,
  [
    Item.Dye_1,
    Item.Dye_2,
    Item.Dye_3,
    Item.Dye_4,
    Item.Dye_5,
    Item.Dye_6,
    Item.Dye_7,
    Item.Dye_8,
    Item.Dye_9,
    Item.Dye_10,
    Item.Dye_11,
  ],
  100_000,
);
