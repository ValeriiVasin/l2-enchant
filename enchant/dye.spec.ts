import { dyeEnchantMap } from './dye';

it('enchant map: dye', () => {
  expect(dyeEnchantMap).toMatchSnapshot();
});
