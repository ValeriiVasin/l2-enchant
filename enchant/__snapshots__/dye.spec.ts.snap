// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: dye 1`] = `
Map {
  "Узор" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Узор Ур.1",
      },
      Object {
        "chance": 50,
        "item": "Узор",
      },
    ],
  },
  "Узор Ур.1" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.1",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Узор Ур.2",
      },
      Object {
        "chance": 75,
        "item": "Узор Ур.1",
      },
    ],
  },
  "Узор Ур.2" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.2",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Узор Ур.3",
      },
      Object {
        "chance": 80,
        "item": "Узор Ур.2",
      },
    ],
  },
  "Узор Ур.3" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.3",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Узор Ур.4",
      },
      Object {
        "chance": 85,
        "item": "Узор Ур.3",
      },
    ],
  },
  "Узор Ур.4" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.4",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Узор Ур.5",
      },
      Object {
        "chance": 90,
        "item": "Узор Ур.4",
      },
    ],
  },
  "Узор Ур.5" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.5",
    "results": Array [
      Object {
        "chance": 8,
        "item": "Узор Ур.6",
      },
      Object {
        "chance": 92,
        "item": "Узор Ур.5",
      },
    ],
  },
  "Узор Ур.6" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.6",
    "results": Array [
      Object {
        "chance": 6,
        "item": "Узор Ур.7",
      },
      Object {
        "chance": 94,
        "item": "Узор Ур.6",
      },
    ],
  },
  "Узор Ур.7" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.7",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Узор Ур.8",
      },
      Object {
        "chance": 95,
        "item": "Узор Ур.7",
      },
    ],
  },
  "Узор Ур.8" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.8",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Узор Ур.9",
      },
      Object {
        "chance": 95,
        "item": "Узор Ур.8",
      },
    ],
  },
  "Узор Ур.9" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.9",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Узор Ур.10",
      },
      Object {
        "chance": 95,
        "item": "Узор Ур.9",
      },
    ],
  },
  "Узор Ур.10" => Object {
    "cost": "Красящий Порошок",
    "item": "Узор Ур.10",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Узор Ур.11",
      },
      Object {
        "chance": 95,
        "item": "Узор Ур.10",
      },
    ],
  },
  "Узор Ур.11" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.11",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Узор Ур.12",
      },
      Object {
        "chance": 90,
        "item": "Узор Ур.11",
      },
    ],
  },
  "Узор Ур.12" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.12",
    "results": Array [
      Object {
        "chance": 8,
        "item": "Узор Ур.13",
      },
      Object {
        "chance": 92,
        "item": "Узор Ур.12",
      },
    ],
  },
  "Узор Ур.13" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.13",
    "results": Array [
      Object {
        "chance": 6,
        "item": "Узор Ур.14",
      },
      Object {
        "chance": 94,
        "item": "Узор Ур.13",
      },
    ],
  },
  "Узор Ур.14" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.14",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Узор Ур.15",
      },
      Object {
        "chance": 95,
        "item": "Узор Ур.14",
      },
    ],
  },
  "Узор Ур.15" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.15",
    "results": Array [
      Object {
        "chance": 4,
        "item": "Узор Ур.16",
      },
      Object {
        "chance": 96,
        "item": "Узор Ур.15",
      },
    ],
  },
  "Узор Ур.16" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.16",
    "results": Array [
      Object {
        "chance": 3,
        "item": "Узор Ур.17",
      },
      Object {
        "chance": 97,
        "item": "Узор Ур.16",
      },
    ],
  },
  "Узор Ур.17" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.17",
    "results": Array [
      Object {
        "chance": 2,
        "item": "Узор Ур.18",
      },
      Object {
        "chance": 98,
        "item": "Узор Ур.17",
      },
    ],
  },
  "Узор Ур.18" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.18",
    "results": Array [
      Object {
        "chance": 1.5,
        "item": "Узор Ур.19",
      },
      Object {
        "chance": 98.5,
        "item": "Узор Ур.18",
      },
    ],
  },
  "Узор Ур.19" => Object {
    "cost": "Улучшенный Красящий Порошок",
    "item": "Узор Ур.19",
    "results": Array [
      Object {
        "chance": 0.5,
        "item": "Узор Ур.20",
      },
      Object {
        "chance": 99.5,
        "item": "Узор Ур.19",
      },
    ],
  },
}
`;
